### PR TITLE
Allow user to make scores public/private

### PR DESCRIPTION
**Pull request by Caleb-Littlejohn**

### CODE REVIEW

File: benchmarks/view.py [hunk 1]
```diff
@@ -13,6 +13,12 @@
 import json
 import datetime
 from .views.index import get_context
+<<<<<<< HEAD
+from django.http import JsonResponse
+from .models import ModelReference
+
+=======
+>>>>>>> 94661307b4e33ac1acea856275bc957f59191d0c
```
Comment (mschrimpf) [inline benchmarks/view.py:21]: you have a git merge remnant here

File: benchmarks/templates/benchmarks/profile-table.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{% load static %}
+
+{% if models %}
+<form>
```
Comment (mschrimpf) [inline benchmarks/templates/benchmarks/profile-table.html:4]: would it be possible to use the same table template as for the main page? I'm worried we'll change that at some point and will run into discrepancies.
(If the difference arises from the checkbox, we could just not display it for the main overview table)

File: static/benchmarks/js/table-public.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+$(document).ready(function(){
+    var checkedCheckboxes = {};
+
+    $('input').click(function(){
+
+    var checkedCheckboxes = {};
+        $('input').each(function(){
+
+            console.log($(this).is(':checked'));
```
Comment (mschrimpf) [inline static/benchmarks/js/table-public.js:9]: I'd remove all the console logs

File: static/benchmarks/js/table-public.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            checkedCheckboxes[$(this).val()] = $(this).is(':checked')
+        });
+
+    // Now we have an array
+    console.log('JS Array: ');
+    console.log(checkedCheckboxes);
```
Comment (mschrimpf) [inline static/benchmarks/js/table-public.js:15]: remove

File: static/benchmarks/js/table-public.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    // Convert array to standard Javascript Object Literal
+    var checkedCheckboxesObject = $.extend({}, checkedCheckboxes);
+    console.log('JS Object: ');
+    console.log(checkedCheckboxesObject);
```
Comment (mschrimpf) [inline static/benchmarks/js/table-public.js:20]: remove

File: static/benchmarks/js/table-public.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+    // Convert Object Literal to JSON
+    var checkedCheckboxesJSON = JSON.stringify(checkedCheckboxesObject);
+    console.log('JSON: ');
+    console.log(checkedCheckboxesJSON);
```
Comment (mschrimpf) [inline static/benchmarks/js/table-public.js:25]: remove

File: static/benchmarks/js/table-public.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+                    cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+                    break;
+                }
+            }
+        }
+        return cookieValue;
+    }
+
+    var csrftoken = getCookie('csrftoken');
+    console.log(csrftoken);
```
Comment (mschrimpf) [inline static/benchmarks/js/table-public.js:44]: remove

File: static/benchmarks/js/table-public.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        }
+    });
+    $.ajax({
+            type: "POST",
+            url: "../public-ajax/",
+            dataType: 'json',
+            contentType: 'json',
+            data: JSON.stringify(checkedCheckboxesJSON),
+            success: function(msg){
+                console.log(msg);
```
Comment (mschrimpf) [inline static/benchmarks/js/table-public.js:65]: remove

File: static/benchmarks/js/table-public.js (hunk 3 above)
Comment (Caleb-Littlejohn) [reply static/benchmarks/js/table-public.js:9]: Removed

File: static/benchmarks/js/table-public.js (hunk 4 above)
Comment (Caleb-Littlejohn) [reply static/benchmarks/js/table-public.js]: Removed

File: static/benchmarks/js/table-public.js (hunk 5 above)
Comment (Caleb-Littlejohn) [reply static/benchmarks/js/table-public.js]: Removed

File: static/benchmarks/js/table-public.js (hunk 6 above)
Comment (Caleb-Littlejohn) [reply static/benchmarks/js/table-public.js]: Removed

File: static/benchmarks/js/table-public.js (hunk 7 above)
Comment (Caleb-Littlejohn) [reply static/benchmarks/js/table-public.js]: Removed

File: static/benchmarks/js/table-public.js (hunk 8 above)
Comment (Caleb-Littlejohn) [reply static/benchmarks/js/table-public.js]: Removed

File: benchmarks/templates/benchmarks/table.html [hunk 9]
```diff
@@ -15,17 +15,17 @@
                     <strong class="want_to_click" style="padding: 5px; display: {{ benchmark_parents|in_set_hidden:benchmark.name }}">{{ benchmark.parent }}:</strong>{% endif %}
                     
                 {% if benchmark.link %}
-                    <a href="{{ benchmark.link }}">{{ benchmark.name }}
-                        {%  if benchmark.version %}
-                        <span class="benchmark-version">v{{ benchmark.version }}</span>
-                        {% endif %}
-                    </a></p>
```
Comment (mschrimpf) [inline benchmarks/templates/benchmarks/table.html:22]: not sure if this is a merge error or intentional, but I think we should leave the version in there (#12)

File: benchmarks/templates/benchmarks/table.html [hunk 10]
```diff
@@ -15,17 +15,17 @@
...
                 {% else %}
                 <span class="want_to_click">{{ benchmark.name }}</span>
 
                 {% endif %}
             </th>
+
             {% endfor %}
+            {% if has_user %}
+            <th><p><span style="padding-left: 5px" class="want_to_click"><b>Public</b></span></p></th>
+            {% endif %}
```
Comment (mschrimpf) [inline benchmarks/templates/benchmarks/table.html:28]: much better!

**Pull request merged**